xbmc/cores/AudioEngine/Sinks/AESinkDARWINIOS.mm

```diff
@@ -532,53 +546,50 @@ static void EnumerateDevices(AEDeviceInfoList &list)
   device.m_deviceName = "default";
   device.m_displayName = "Default";
   device.m_displayNameExtra = "";
-  // TODO screen changing on ios needs to call
-  // devices changed once this is available in active
-  if (0)
-  {
-    device.m_deviceType = AE_DEVTYPE_IEC958; //allow passthrough for tvout
-    device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_AC3);
-    device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTSHD_CORE);
-    device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTS_2048);
-    device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTS_1024);
-    device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTS_512);
-    device.m_dataFormats.push_back(AE_FMT_RAW);
-  }
-  else
-    device.m_deviceType = AE_DEVTYPE_PCM;
 
-  // add channel info
-  CAEChannelInfo channel_info;
-  for (UInt32 chan = 0; chan < 2; ++chan)
-  {
-    if (!device.m_channels.HasChannel(CAChannelMap[chan]))
-      device.m_channels += CAChannelMap[chan];
-    channel_info += CAChannelMap[chan];
-  }
+  // if not hdmi,  CAESinkDARWINIOS::Initialize will kick back to 2 channel PCM
+  device.m_deviceType = AE_DEVTYPE_HDMI;
+  device.m_wantsIECPassthrough = true;
+
+  device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_AC3);
+  device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_EAC3);
+  device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTS_512);
+  device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTS_1024);
+  device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTS_2048);
+  device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTSHD_CORE);
+  // ATV can not do below (yet :)
+  // device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTSHD);
+  // device.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_TRUEHD);
```

Comment:
#ifndef (TARGET_DARWIN_TVOS) ??
do the following lines work with ios? if not, drop it all?
or if its atv specific, bundle in an ifdef target_darwin_tvos

May be better to drop all tvos specific code in this PR and add to the ATV Pr as a followup instead

Reply:
remove, tvOS can not do dts-hd or truehd. It was just a placeholder just-in-case.